server/middlewares/adminAuth.ts

```diff
@@ -0,0 +1,15 @@
+import { Response, NextFunction } from 'express';
+import { CustomError } from '.';
+import { UserRequest } from './interfaces';
+
+const pharmacyAuth = async (req: UserRequest['user'], res: Response, next: NextFunction) => {
```

Comment:
Is it a middleware to check the admin or the pharmacy? match the func name and the role